contracts/nft/contracts/solana/programs/universal-nft/src/lib.rs

```diff
@@ -0,0 +1,743 @@
+                Some(signer_seeds),
+            )?;
+        }
+
+        // 2) Load or create origin account backed by these accounts and the provided URI.
+        let mut origin_data: NftOrigin;
+        if ctx.accounts.nft_origin.to_account_info().owner == &crate::id() && !ctx.accounts.nft_origin.data_is_empty() {
+            // Manually deserialize existing origin account to avoid lifetime issues
+            let data_ref = ctx.accounts.nft_origin.try_borrow_data()?;
+            require!(data_ref.len() >= 8, UniversalNftError::InvalidDataFormat);
+            let mut bytes: &[u8] = &data_ref[8..]; // skip discriminator
+            origin_data = NftOrigin::try_deserialize(&mut bytes)
+                .map_err(|_| UniversalNftError::InvalidDataFormat)?;
+        } else {
+            origin_data = NftOrigin::new(
+                token_id,
+                ctx.accounts.mint_account.key(),
+                ctx.accounts.metadata.key(),
+                metadata_uri.clone(),
+                clock.unix_timestamp,
+                origin_bump,
+            );
+
+            let rent = Rent::get()?;
+            let space = NftOrigin::SPACE as u64;
+            let lamports = rent.minimum_balance(space as usize);
+            
+            // Sign as both payer PDA and the new nft_origin PDA
+            let config_seeds_raw: &[&[u8]] = &[UNIVERSAL_NFT_CONFIG_SEED, &[ctx.accounts.pda.bump]];
+            let nft_origin_seeds: &[&[u8]] = &[
+                NFT_ORIGIN_SEED,
+                &token_id,
+                &[origin_bump],
+            ];
+            let create_signers: &[&[&[u8]]] = &[&config_seeds_raw[..], nft_origin_seeds];
+
+            anchor_lang::system_program::create_account(
+                CpiContext::new_with_signer(
+                    ctx.accounts.system_program.to_account_info(),
+                    anchor_lang::system_program::CreateAccount {
+                        from: ctx.accounts.pda.to_account_info(),
+                        to: ctx.accounts.nft_origin.to_account_info(),
+                    },
+                    create_signers,
+                ),
+                lamports,
+                space,
+                &crate::id(),
+            )?;
+
+            let mut data_ref = ctx.accounts.nft_origin.try_borrow_mut_data()?;
+            
+            // Write full account (discriminator + data)
+            let mut tmp = Vec::with_capacity(NftOrigin::SPACE);
+            anchor_lang::prelude::AccountSerialize::try_serialize(&origin_data, &mut tmp)
+                .map_err(|_| UniversalNftError::InvalidDataFormat)?;
+            require!(tmp.len() <= NftOrigin::SPACE, UniversalNftError::InvalidDataFormat);
+            data_ref.fill(0);
+            data_ref[..tmp.len()].copy_from_slice(&tmp);
+        }
+
+        // 3) Mint 1 to program_ata using config PDA as authority
+        mint_nft_to_recipient(
+            &ctx.accounts.mint_account,
+            &ctx.accounts.pda_ata,
```

Comment:
How does this get trasferd to the user who initiated the transfer ?

Reply:
what is the issue you were facing ? 
Its not super clear to me what you mean by not using remianing accounts,  
the account_metas for the on_call is created like so 

https://github.com/zeta-chain/protocol-contracts-solana/blob/f5d96872749864e4ae98241455cd1336f0fae90d/programs/gateway/src/utils/prepare_account_metas.rs#L8-L32